File: Python/0696.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        ans = 0
        prev = 0
        curr = 1

        for i in range(1, len(s)):
            if s[i - 1] != s[i]:
                ans += min(prev, curr)
                prev = curr
                curr = 1
            else:
                curr += 1

        return ans + min(prev, curr)

File: C/0696.c
#include <string.h>

int min(int a, int b);

int countBinarySubstrings(char* s) {
    int ans = 0, prev = 0, curr = 1;
    int n = strlen(s);

    for (int i = 1; i < n; i++) {
        if (s[i - 1] != s[i]) {
            ans += min(prev, curr);
            prev = curr;
            curr = 1;
        } else {
            curr++;
        }
    }

    return ans + min(prev, curr);
}

int min(int a, int b) {
    if (a < b) {
        return a;
    }

    return b;
}
